src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -73,62 +73,51 @@ public unsafe bool ParseRequestLine(TRequestHandler handler, in ReadOnlySequence
         private unsafe void ParseRequestLine(TRequestHandler handler, byte* data, int length)
         {
             // Get Method and set the offset
-            var method = HttpUtilities.GetKnownMethod(data, length, out var offset);
+            var method = HttpUtilities.GetKnownMethod(data, length, out var pathStartOffset);
 
-            Span<byte> customMethod = method == HttpMethod.Custom ?
-                GetUnknownMethod(data, length, out offset) :
-                default;
+            Span<byte> customMethod = default;
+            if (method == HttpMethod.Custom)
+            {
+                customMethod = GetUnknownMethod(data, length, out pathStartOffset);
+            }
 
+            // Use a new offset var as pathStartOffset needs to be on stack
+            // as its passed by reference above so can't be in register.
```

Comment:
Can you explain this a little more? Why is it important to use pathStartOffset for the out var instead of reusing offset like before? Is there better performance if you don't mutate the out var or something?

Reply:
Its pass byref so it needs a memory location; registers are only used for parameters following [strict ABI rules](https://github.com/dotnet/coreclr/blob/master/Documentation/botr/clr-abi.md).

Basically the location will get passed via register; but it needs a location on stack to be that location. Which is fine.

The problem is the Jit will only allocate the local var once. So it will get a stack location rather than a register, because it needs to be passed byref.

The side effect is every other use of that local in the function will also be that stack location; so you get this:
```asm
mov      ecx, dword ptr [rbp-40H]
inc      ecx
mov      dword ptr [rbp-40H], ecx
cmp      dword ptr [rbp-40H], edi
jl       SHORT G_M24537_IG10
```
With the memory location `dword ptr [rbp-40H]` being the local variable

Assigning it to another variable an then using that one instead breaks the dependency on that memory location, so instead you get this:
```asm
inc      r8d
cmp      r8d, edi
jl       SHORT G_M24517_IG13 
```
With the register `r8d` being the local variable